src/payment/strategies/paypal-commerce/paypal-commerce-payment-processor.ts

```diff
@@ -0,0 +1,62 @@
+import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+
+import { OverlayPaypal } from './index';
+
+const modalWidth = 450;
+const modalHeight = 600;
+
+export default class PaypalCommercePaymentProcessor {
+    private _window = window;
+    private _popup?: Window | null;
+
+    constructor(
+        private _overlay: OverlayPaypal
+    ) {}
+
+    paymentPayPal(approveUrl: string): Promise<any> {
+        return new Promise((resolve, reject) => {
+            const closeWindow = (isResolve: boolean) => {
```

Comment:
instead of defining functions inside a function, could you instead define them as private functions?

`_closeWindow`
`_ handlerMessage`

Reply:
Hm, thank you for your suggest!
I tried to do it, but thought it's not so elegantly

Reply:
I implemented this code, but it doesn't work well. In my implementation after `reject` user sees standard modal window with error and in your suggestion modal window doesn't showed. So I prefer to leave my code